src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -194,7 +211,46 @@ public int CountTokens(ReadOnlySpan<char> text, bool considerPreTokenization = t
         /// If <paramRef name="fromEnd" /> is <see langword="true"/>, it represents the index of the first character to be included. In cases where no tokens fit, the result will be the text length; conversely,
         /// if all tokens fit, the result will be zero.
         /// </returns>
-        protected abstract int GetIndexByTokenCount(string? text, ReadOnlySpan<char> textSpan, EncodeSettings settings, bool fromEnd, out string? normalizedString, out int tokenCount);
+        protected virtual int GetIndexByTokenCount(string? text, ReadOnlySpan<char> textSpan, EncodeSettings settings, bool fromEnd, out string? normalizedString, out int tokenCount)
+        {
+            int maxTokenCount = settings.MaxTokenCount;
+
+            settings.MaxTokenCount = int.MaxValue;
```

Comment:
>settings.MaxTokenCount = int.MaxValue; [](http://example.com/codeflow?start=12&length=38)

should we do that only if we have `fromEnd = true`?

Reply:
Good call.